### DISCUSSION TITLE
Bootstrap-vue-next tags not converted to proper HTML tags

### DISCUSSION THREAD

**Discussion by yingfeng-iu** [cdmoro/bootstrap-vue-3#2310] [Q&A] [answered]
I'm working on upgrading our AMP project from Vue2 to Vue3, and ran into some blocking issue when switching from Bootstrap-vue to bootstrap-vue-next (BSVN). Basically, none of the components such as b-modal, b-card, b-button gets converted into HTML tags in the generated DOM when the app is running - they just stay as the original bootstrap tags and thus ignored by browser. Here is the link to the project branch using vue 3 and bootstrap-vue-next: https://github.com/AudiovisualMetadataPlatform/amppd-ui/tree/AMP-3426_BSVN. In particular, below are some high-level facts and code snippets:

- https://github.com/AudiovisualMetadataPlatform/amppd-ui/blob/AMP-3426_BSVN/package.json
	"bootstrap": "^5.3.3",
    "bootstrap-vue-next": "^0.24.21",
	"vue": "^3.4.34",
	"@vue/cli": "^5.0.8",
- https://github.com/AudiovisualMetadataPlatform/amppd-ui/blob/AMP-3426_BSVN/src/main.js: the code does include the below lines, following the instruction on bootstrap-vue-next documentation page
	import {createBootstrap} from "bootstrap-vue-next";
	app.use(createBootstrap());
- the project is written in Options API instead of Composition API
- I couldn't figure out how to do the automatic components registration - tried various ways following the documentation but all result in some compiler error. So, instead, I tried to import the referenced BSVN components manually in one of our components https://github.com/AudiovisualMetadataPlatform/amppd-ui/blob/AMP-3426_BSVN/src/components/entity/EntityList.vue, but that didn't make any difference either. The components such as b-card etc still appear in the generated DOM.

As an attempt to figure out how BSVN should work, I found some old examples on https://codesandbox.io/examples/package/bootstrap-vue-next, and forked one project that actually works, but using a much older version of bootstrap-vue-next. When I switched the version to any recent one such as 0.25.10, I got compiler error "e.mergeModels is not a function". here is the forked project: https://codesandbox.io/p/sandbox/vue3-bootstrap-modal-forked-7vpg6n?file=%2Fsrc%2Fmain.js%3A1%2C18&workspaceId=c678e24e-8e67-43ba-816b-03f6099f28ef.

Could someone help me on this? I guess I'm missing some global setup because none of the BSVN component works in either our AMP project, or my sandbox example. Also, does the latest version of BSVN support Options API?

**Comment by VividLemon** [comment, marked as answer]
https://github.com/AudiovisualMetadataPlatform/amppd-ui/pull/543

**Reply by yingfeng-iu** [reply]
Many thanks to you! I merged your PR and it worked like a magic. The best part is, previously, we had to overwrite Bootstrap 5 CSS with an older 4 version, as otherwise the display would be off on many pages. Now with bootstrap-vue-next properly integrated after the PR, I can switch to BS 5 and many pages now look much better! I'm glad that I posted the question here as a last resort (should have done this much sooner). Your timely response saved me many more hours of troubleshooting. We would probably decide to switch to some other framework if we can't get this working. I still have more issues to resolve with some individual BS components. With your help and help from this community, I feel more confident. Thanks again!

**Reply by VividLemon** [reply]
Np
